lib/project/nativescript-project.ts

```diff
@@ -182,6 +184,18 @@ export class NativeScriptProject extends FrameworkProjectBase implements Project
 			}
 		}).future<void>()();
 	}
+
+	public ensureProject(projectDir: string): IFuture<void> {
+		return (() => {
+			if (this.$typeScriptService.isTypeScriptProject(projectDir).wait()) {
+				try {
+					this.$npmService.install(projectDir).wait();
+				} catch (err) {
+					this.$logger.trace(`Failed to install all npm dependencies in the project. Error: ${err}.`);
```

Comment:
If there is an exception, can we state that the project has all the required files? Shouldn't we rethrow the exception?